tuplex/codegen/include/FunctionRegistry.h

```diff
@@ -139,6 +177,7 @@ namespace tuplex {
         private:
             LLVMEnvironment& _env;
             bool _sharedObjectPropagation;
+            IteratorContextProxy *_icp;
```

Comment:
std::shared_ptr? std::unique_ptr? how is the memory here managed?

Reply:
Changed to std::shared_ptr.